pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/AssemblyRendering.java

```diff
@@ -0,0 +1,108 @@
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.redhat.pantheon.conf.GlobalConfig.DEFAULT_MODULE_LOCALE;
+import static com.redhat.pantheon.model.module.ModuleVariant.DEFAULT_VARIANT_NAME;
+import static com.redhat.pantheon.servlet.ServletUtils.paramValue;
+import static com.redhat.pantheon.servlet.ServletUtils.paramValueAsBoolean;
+
+/**
+ * Renders an HTML preview for a single module.
+ * To provide parameters to the asciidoc generation process, provide the parameters with their name prefixed
+ * with "ctx_".
+ *
+ * For example, if an asciidoc attribute of name 'product' needs to be passed, there will need to be a
+ * query parameter of name 'ctx_product' provided in the url.
+ */
+
+
+public class AssemblyRendering implements Rendering {
+
+    private final Logger log = LoggerFactory.getLogger(AssemblyRendering.class);
+
+    static final String PARAM_RERENDER = "rerender";
+    static final String PARAM_DRAFT = "draft";
+    static final String PARAM_LOCALE = "locale";
+    static final String PARAM_VARIANT = "variant";
+
+    private AsciidoctorService asciidoctorService;
+
+    @Activate
+    public AssemblyRendering(
+            @Reference AsciidoctorService asciidoctorService) {
+        this.asciidoctorService = asciidoctorService;
+    }
+
+    @Override
+    public void getRenderedHTML(SlingHttpServletRequest request,
+            SlingHttpServletResponse response) throws IOException {
+        String locale = paramValue(request, PARAM_LOCALE, DEFAULT_MODULE_LOCALE.toString());
+        boolean draft = paramValueAsBoolean(request, PARAM_DRAFT);
+        boolean reRender = paramValueAsBoolean(request, PARAM_RERENDER);
+        String variantName = paramValue(request, PARAM_VARIANT, DEFAULT_VARIANT_NAME);
+
+        Module module = request.getResource().adaptTo(Module.class);
+        Locale localeObj = LocaleUtils.toLocale(locale);
+
+        Optional<HashableFileResource> moduleVariantSource;
+
+        if(draft) {
+            moduleVariantSource = module.moduleLocale(localeObj)
+                .traverse()
+                .toChild(ModuleLocale::source)
+                .toChild(SourceContent::draft)
+                .getAsOptional();
+        } else {
+            moduleVariantSource = module.moduleLocale(localeObj)
+                    .traverse()
+                    .toChild(ModuleLocale::source)
+                    .toChild(SourceContent::released)
+                    .getAsOptional();
+        }
+
+
+        if(!moduleVariantSource.isPresent()) {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND, (draft ? "Draft " : "Released ")
+                    + "source content not found for " + variantName +  " module variant at "
+                    + request.getResource().getPath());
```

Comment:
Rather than having an else you can simply return

Reply:
resolved

Reply:
I don't think this is resolved. It looks like you're missing a `return;` statement after the call to `response.sendError`.